server/pages/index.js

```diff
@@ -111,6 +120,14 @@ function getCurrentCommitShortChecksum() {
 }
 
 function getDefaultContext( request ) {
+	let initialServerState = {};
+	// We don't cache routes with query params
+	if ( isEmpty( request.query ) ) {
```

Comment:
Seems to me that this decision should be left up to the route, but I think we can go with this for now.